mat64/doc.go

```diff
@@ -112,7 +112,47 @@
 // For convenience, a matrix may be used as both a receiver and as an input, i.e.
 //  a.Pow(a, 6)
 //  v.SolveVec(a.T(), v)
-// though in many cases this will cause an allocation. However, mat64 cannot
-// detect arbitrary overlap, and so care should be taken when using matrix views.
+// though in many cases this will cause an allocation (see Element Aliasing).
 // An exception to this rule is Copy, which does not allow a.Copy(a.T()).
+//
+// Element Aliasing
+//
+// Most methods in mat64 modify receiver data. It is forbidden for the modified
+// data region of the receiver to overlap the used data area of the input
+// arguments. The exception to this rule is when the method receiver is equal to one
+// of the input arguments, as in the a.Pow(a, 6) call above, or its implicit transpose.
+//
+// This prohibition is to help avoid subtle mistakes resulting from input corruption
```

Comment:
'subtle mistakes when the method needs to read from and write to the same data region. There are many ways ...'